test/spec/messageEvents.spec.js

```diff
@@ -57,16 +63,18 @@ describe('Message Events Library', function() {
         per_page: 5,
         timezone: 'America/New_York'
       };
-      messageEvents.search(arroptions, function(err, data) {
-        Object.keys(arroptions).forEach(function(key) {
-          var opt = arroptions[key]
-            , firstCallQS = client.get.firstCall.args[0].qs;
-          if (Array.isArray(opt)) {
-            expect(firstCallQS).to.have.property(key).and.equal(opt.toString());
-          }
+      messageEvents.search(arroptions)
+        .then(function() {
+          let tests = [];
+          Object.keys(arroptions).forEach(function(key) {
+            var opt = arroptions[key]
+              , firstCallQS = client.get.firstCall.args[0].qs;
+            if (Array.isArray(opt)) {
+              tests.push(expect(firstCallQS).to.have.property(key).and.equal(opt.toString()));
```

Comment:
these are synchronous expectations, so no need to wrap them in the promise later. Expect will throw an error if the assertion fails, and that will result in a rejected promise since we are inside a `then` callback`.